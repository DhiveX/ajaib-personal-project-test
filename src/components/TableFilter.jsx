import React from "react";

export default function TableFilter(props) {
  function handleFilter(value) {
    const temporaryFilterList = { ...value };
    props.setFilter(temporaryFilterList);
  }

  function handleKeyword(event) {
    let tempValue = props.appliedFilter;
    tempValue.keyword = event.target.value;
    handleFilter(tempValue);
    props.applyFilter();
  }

  function handleGender(event) {
    let tempValue = props.appliedFilter;
    tempValue.gender = event.target.value;
    handleFilter(tempValue);
  }

  function handleSortBy(event) {
    let tempValue = props.appliedFilter;
    tempValue.sortBy = event.target.value;
    handleFilter(tempValue);
  }
  function handleSortOrder(event) {
    let tempValue = props.appliedFilter;
    tempValue.sortOrder = event.target.value;
    handleFilter(tempValue);
  }

  const handleResetFilter = () => {
    let tempValue = props.appliedFilter;
    tempValue.page = 1;
    tempValue.pageSize = 10;
    tempValue.gender = null;
    tempValue.keyword = null;
    tempValue.sortBy = null;
    tempValue.sortOrder = "ascend";
    tempValue.results = 30;
    tempValue.seed = "ajaib";
    handleFilter(tempValue);
    document.getElementById("formFilter").reset();
    props.applyFilter();
  };
  return (
    <div className="mb-6">
      <form className="flex gap-[20px] flex-wrap justify-end" id="formFilter">
        <input
          type="text"
          placeholder="I'm a Search Bar"
          onChange={handleKeyword}
          className="rounded-lg !border-white-700 border-2 p-2 shadow-xl"
        ></input>
        <select
          onChange={handleGender}
          className="rounded-lg !border-white-700 border-2 p-2 shadow-xl text-gray-500"
        >
          <option value="">--select gender--</option>
          <option value="female">female</option>
          <option value="male">male</option>
        </select>
        <select
          onChange={handleSortBy}
          className="rounded-lg !border-white-700 border-2 p-2 shadow-xl text-gray-500"
        >
          <option value="">--select sort by--</option>
          <option value="username">username</option>
          <option value="name">name</option>
          <option value="email">email</option>
          <option value="gender">gender</option>
          <option value="registeredDate">registered date</option>
        </select>
        <select
          onChange={handleSortOrder}
          className="rounded-lg !border-white-700 border-2 p-2 shadow-xl text-gray-500"
        >
          <option value="">--select sort order--</option>
          <option value="ascend">ascending</option>
          <option value="descend">descending</option>
        </select>
        <button
          className="rounded-lg text-blue-600 !border-white-700 border-2 p-2 shadow-xl bg-white"
          type="button"
          onClick={() => props.applyFilter()}
        >
          Apply Filter
        </button>
        <button
          className="rounded-lg text-blue-600 !border-white-700 border-2 p-2 shadow-xl bg-white"
          type="button"
          onClick={() => handleResetFilter()}
        >
          Reset Filter
        </button>
      </form>
    </div>
  );
}
