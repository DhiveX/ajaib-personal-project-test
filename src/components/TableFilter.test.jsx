import { render } from "@testing-library/react";
import TableFilter from "./TableFilter";

describe(TableFilter, () => {
  it("Filter get Visible to User", () => {
    const { getByTestId } = render(<TableFilter />);
    const tableFilter = getByTestId("tableFilter");
    expect(tableFilter).toBeVisible();
  });

  it("should render with filter form", () => {
    const component = render(<TableFilter />);
    expect(component).toMatchSnapshot();
  });
});
