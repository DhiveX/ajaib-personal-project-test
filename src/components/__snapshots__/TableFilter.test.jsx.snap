// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableFilter should render with filter form 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mb-6"
        data-testid="tableFilter"
      >
        <form
          class="flex gap-[20px] flex-wrap justify-end"
          id="formFilter"
        >
          <div
            class="flex-col flex"
          >
            <label
              class="text-white font-bold"
            >
              Seach
            </label>
            <input
              class="rounded-lg !border-white-700 border-2 p-2 shadow-xl"
              placeholder="I'm a Search Bar"
              type="text"
            />
          </div>
          <div
            class="flex-col flex"
          >
            <label
              class="text-white font-bold"
            >
              Filter By Gender
            </label>
            <select
              class="rounded-lg !border-white-700 border-2 p-2 shadow-xl text-gray-500"
            >
              <option
                value=""
              >
                --select gender--
              </option>
              <option
                value="female"
              >
                female
              </option>
              <option
                value="male"
              >
                male
              </option>
            </select>
          </div>
          <div
            class="flex-col flex"
          >
            <label
              class="text-white font-bold"
            >
              Sort By
            </label>
            <select
              class="rounded-lg !border-white-700 border-2 p-2 shadow-xl text-gray-500"
            >
              <option
                value=""
              >
                --select sort by--
              </option>
              <option
                value="username"
              >
                username
              </option>
              <option
                value="name"
              >
                name
              </option>
              <option
                value="email"
              >
                email
              </option>
              <option
                value="gender"
              >
                gender
              </option>
              <option
                value="registeredDate"
              >
                registered date
              </option>
            </select>
          </div>
          <div
            class="flex-col flex"
          >
            <label
              class="text-white font-bold"
            >
              Sort Order
            </label>
            <select
              class="rounded-lg !border-white-700 border-2 p-2 shadow-xl text-gray-500"
            >
              <option
                value=""
              >
                --select sort order--
              </option>
              <option
                value="ascend"
              >
                ascending
              </option>
              <option
                value="descend"
              >
                descending
              </option>
            </select>
          </div>
          <button
            class="rounded-lg text-blue-600 !border-white-700 border-2 p-2 shadow-xl bg-white"
            data-testid="applyFilterButton"
            type="button"
          >
            Apply Filter
          </button>
          <button
            class="rounded-lg text-blue-600 !border-white-700 border-2 p-2 shadow-xl bg-white"
            data-testid="resetFilterButton"
            type="button"
          >
            Reset Filter
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mb-6"
      data-testid="tableFilter"
    >
      <form
        class="flex gap-[20px] flex-wrap justify-end"
        id="formFilter"
      >
        <div
          class="flex-col flex"
        >
          <label
            class="text-white font-bold"
          >
            Seach
          </label>
          <input
            class="rounded-lg !border-white-700 border-2 p-2 shadow-xl"
            placeholder="I'm a Search Bar"
            type="text"
          />
        </div>
        <div
          class="flex-col flex"
        >
          <label
            class="text-white font-bold"
          >
            Filter By Gender
          </label>
          <select
            class="rounded-lg !border-white-700 border-2 p-2 shadow-xl text-gray-500"
          >
            <option
              value=""
            >
              --select gender--
            </option>
            <option
              value="female"
            >
              female
            </option>
            <option
              value="male"
            >
              male
            </option>
          </select>
        </div>
        <div
          class="flex-col flex"
        >
          <label
            class="text-white font-bold"
          >
            Sort By
          </label>
          <select
            class="rounded-lg !border-white-700 border-2 p-2 shadow-xl text-gray-500"
          >
            <option
              value=""
            >
              --select sort by--
            </option>
            <option
              value="username"
            >
              username
            </option>
            <option
              value="name"
            >
              name
            </option>
            <option
              value="email"
            >
              email
            </option>
            <option
              value="gender"
            >
              gender
            </option>
            <option
              value="registeredDate"
            >
              registered date
            </option>
          </select>
        </div>
        <div
          class="flex-col flex"
        >
          <label
            class="text-white font-bold"
          >
            Sort Order
          </label>
          <select
            class="rounded-lg !border-white-700 border-2 p-2 shadow-xl text-gray-500"
          >
            <option
              value=""
            >
              --select sort order--
            </option>
            <option
              value="ascend"
            >
              ascending
            </option>
            <option
              value="descend"
            >
              descending
            </option>
          </select>
        </div>
        <button
          class="rounded-lg text-blue-600 !border-white-700 border-2 p-2 shadow-xl bg-white"
          data-testid="applyFilterButton"
          type="button"
        >
          Apply Filter
        </button>
        <button
          class="rounded-lg text-blue-600 !border-white-700 border-2 p-2 shadow-xl bg-white"
          data-testid="resetFilterButton"
          type="button"
        >
          Reset Filter
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
